Frameworks/Foundation/NSKVOSupport.mm

```diff
@@ -226,6 +213,45 @@ - (bool)isEmpty {
                                               NSArray* affectedObservers);
 static void _removeKeyObserver(_NSKVOKeyObserver* keyObserver);
 
+// Private helper that backs the default implementation of keyPathsForValuesAffectingValueForKey:
+// Returns nil instead of constructing an empty set if no keyPaths are found
+// Also used internally as a minor optimization, to avoid constructing an empty set when it is not needed
+static NSSet* _keyPathsForValuesAffectingValueForKey(Class self, NSString* key) {
```

Comment:
I think we have other instances of 'self' being what it would normally be, were this an objc function. Wontfixing this.